lib/spack/spack/directory_layout.py

```diff
@@ -349,6 +351,15 @@ def remove_install_directory(self, spec, deprecated=False):
         path = self.path_for_spec(spec)
         assert(path.startswith(self.root))
 
+        # Windows readonly files cannot be removed by Python
+        # directly. error_remove_readonly respects ignore_error
+        # as True in general while handling the readonly case
```

Comment:
> respects ignore_error as True

I think that here we generally want to `raise` if the error isn't a permissions error. To that end, `error_remove_readonly` can `raise` if the error isn't the one this PR prepares to handle.

That would mean that `remove_linked_tree` would then generally behave as `ignore_errors=False` on Windows. This would conflict with the test added in https://github.com/spack/spack/pull/29690 (although that test already doesn't run on Windows) but shouldn't cause a regression in the actual fix because that PR also updated how `remove_linked_tree` was called.

Another option is to customize `fs.error_remove_readonly` as a callable object (i.e. make it a `ErrorRemoveReadonly` class with a `__call__` method that either `raise`es or `pass`es depending on how the object was instantiated).

Reply:
The method has been amended. I liked the callable approach and now a method configured with `ignore_errors` will return a function pointer to the  `error_remove_readonly` method with the `ignore_errors` value set as per the initial call. Let me know if the docstring is unclear.